From bb775f14e37a4eab014416fe7cc92872302fcc9b Mon Sep 17 00:00:00 2001
From: Yinhang Liu <yinhang.liu@intel.com>
Date: Tue, 25 Apr 2023 16:03:08 +0800
Subject: [PATCH 116/205] msdkmjpegdec: Fix ColorFormat for BGRx format

For msdkmjpegdec, the BGRx format is supported in src caps,
set the ColorFormat to MFX_JPEG_COLORFORMAT_RGB.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/4489>
---
 sys/msdk/gstmsdkcaps.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sys/msdk/gstmsdkcaps.c b/sys/msdk/gstmsdkcaps.c
index d23db0c28..297fa94f6 100644
--- a/sys/msdk/gstmsdkcaps.c
+++ b/sys/msdk/gstmsdkcaps.c
@@ -917,6 +917,7 @@ _jpegdec_set_color_format (mfxVideoParam * param, GstVideoFormat format)
       param->mfx.JPEGColorFormat = MFX_JPEG_COLORFORMAT_YCbCr;
       break;
     case GST_VIDEO_FORMAT_BGRA:
+    case GST_VIDEO_FORMAT_BGRx:
       param->mfx.JPEGColorFormat = MFX_JPEG_COLORFORMAT_RGB;
       break;
     default:
-- 
2.40.1

